### DISCUSSION TITLE
ThreadPoolExecutor + httpx.AsyncClient: do I need a singe AsyncClient or one per thread?

### DISCUSSION THREAD

**Discussion by ok512** [encode/httpx#3074] [Q&A]
The reason I'm asking:
I'm getting exception
ThreadPoolExecuton _eventloop.py _asyncio.py AttributeError partially initialized module has no attribute 'backend_class'
in ~5% of the runs with "each thread has its own httpx.AsyncClient"

I didn't measure singe AsyncClient yet...

**Comment by lovelydinosaur** [comment]
If you've got multiple threads, and you're running async within each, then yes use a client within each thread.
The pooling with `Client` is designed to be thread-safe, the pooling with `AsyncClient` is designed to be task-safe.
Mixing the two isn't a use-case I'd anticipate, so we could feasibly document this.
